packages/mongo/mongo_driver.js

```diff
@@ -907,12 +907,27 @@ function setupSynchronousCursor(cursor, method) {
   return cursor._synchronousCursor;
 }
 
+
 [...ASYNC_CURSOR_METHODS, Symbol.iterator, Symbol.asyncIterator].forEach(methodName => {
+  // count is handled specially since we don't want to create a cursor.
+  // it is still included in ASYNC_CURSOR_METHODS because we still want an async version of it to exist.
+  if (methodName === 'count') {
+    return;
+  }
   Cursor.prototype[methodName] = function (...args) {
     const cursor = setupSynchronousCursor(this, methodName);
     return cursor[methodName](...args);
   };
 
+
+  Cursor.prototype.count = function () {
```

Comment:
This should go outside of the `forEach` loop, or inside the `methodName === 'count'` guard, so `Cursor.prototype.count` is not redefined on every iteration of the loop.

Reply:
:facepalm: thanks for the catch!